nilearn/glm/second_level/second_level.py

```diff
@@ -773,12 +843,38 @@ def non_parametric_inference(second_level_input, confounds=None,
     target_vars = masker.transform(effect_maps)
 
     # Perform massively univariate analysis with permuted OLS
-    neg_log_pvals_permuted_ols, _, _ = permuted_ols(
-        tested_var, target_vars, model_intercept=model_intercept,
-        n_perm=n_perm, two_sided_test=two_sided_test,
-        random_state=random_state, n_jobs=n_jobs,
-        verbose=max(0, verbose - 1))
-    neg_log_corrected_pvals_img = masker.inverse_transform(
-        np.ravel(neg_log_pvals_permuted_ols))
-
-    return neg_log_corrected_pvals_img
+    outputs = permuted_ols(
+        tested_var,
+        target_vars,
+        model_intercept=model_intercept,
+        n_perm=n_perm,
+        two_sided_test=two_sided_test,
+        random_state=random_state,
+        n_jobs=n_jobs,
+        verbose=max(0, verbose - 1),
+        masker=masker,
+        threshold=threshold,
+        output_type='dict',
+    )
+    neg_log10_vfwe_pvals = outputs['logp_max_t']
+    neg_log10_vfwe_pvals_img = masker.inverse_transform(
+        np.ravel(neg_log10_vfwe_pvals),
+    )
+
+    if threshold is not None:
+        t_img = masker.inverse_transform(np.ravel(outputs['t']))
+        neg_log10_csfwe_pvals_img = masker.inverse_transform(
```

Comment:
what is `csfwe` short for ? I saw `vfwe` defined in the doc, but it would be nice to spell out these abbreviations (at least once) in the comments to help orient folks to the code !

Reply:
It stands for cluster size-based FWE-corrected. I can add a comment.

Reply:
I added a few comments in [d98d26e](https://github.com/nilearn/nilearn/pull/3181/commits/d98d26e9d443ddda833523e7bb1dae34259e2fb0). WDYT?